### DISCUSSION TITLE
API Return ID if the contact is already registered

### DISCUSSION THREAD

**Discussion by jeovajireh** [chatwoot/chatwoot#1574] [Q&A]
Hello everybody!

  I need help with an API return message. I need that when registering a contact if the email or phone is already registered it will return the contact ID and not an error message. What are the files that handle the contact record and error messages?

Thanks

**Comment by pranavrajs** [comment]
@viapratica There is a contact search API https://www.chatwoot.com/developers/api/#operation/contactSearch. Let me know if that helps.

**Reply by jeovajireh** [reply]
Yes you are right. 

I want to take the opportunity to ask another question and if it is the case to open a new question, let me know.
Our old chat had a menu with the options of "inbox", Chatwoot requires that when creating the contact an inbox must be defined, I tried to use an "*" but it does not generate the source_id and if the contact needs support in another department (inbox) he has to create another contact, the point is that Chatwoot will not accept the same email and phone number of the client for that other contact. Is there how to create a contact for all inbox?

Sorry for bad English, if you don't understand something, let me know.

Thank you very much for your support

**Reply by sojan-official** [reply]
@viapratica  You could create inboxes for multiple departments and then associate the contact to the appropriate inbox by using the contact inbox api

https://www.chatwoot.com/developers/api/#operation/contactInboxCreation

fetching a contact will return all the associated contact inboxes. you could then use that to create the conversation under the specific inbox. Let me if this helps.

**Reply by jeovajireh** [reply]
Unfortunately in charge

  https://www.chatwoot.com/developers/api/#operation/contactInboxCreation

I got lost and couldn't use it, tried several combinations, but in all of them, either the error page or {"error": "Resource could not be found"}

excuse me

**Reply by jeovajireh** [reply]
curl --request POST 'http://127.0.0.1:3003/api/v1/accounts/1/contacts/19/contact_inboxes/'  --header 'Content-Type: application/json' --header 'api_access_token: NQfhAnM58ThYybXV119BYrLP'

{"error":"Resource could not be found"}

curl --request POST 'http://127.0.0.1:3003/api/v1/accounts/1/contacts/19/contact_inboxes/1'  --header 'Content-Type: application/json' --header 'api_access_token: NQfhAnM58ThYybXV119BYrLP'

<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8" />
  <title>Action Controller: Exception caught</title>
  <style>
....

**Reply by jeovajireh** [reply]
> @viapratica There is a contact search API https://www.chatwoot.com/developers/api/#operation/contactSearch. Let me know if that helps.

There is a problem with this part of the documentation, it doesn't mention any **parameters**, so I couldn't get it to work.

"
Create contact inbox
Create a contact inbox record for an inbox

AUTHORIZATIONS: userApiKey
Responses
"

I was only able to use the command after checking this file in the code.

https://github.com/chatwoot/chatwoot/blob/f61b731fb39aab006a41522f5195a1eed20ac6eb/swagger/paths/contact_inboxes/create.yml

Where I found the measurement to the parameter that must be passed.

   parameters:
     - **name: inbox_id**
       in: params
       type: number
   responses:
     200:
       description: Success


The correct is:

curl --request POST 'http://127.0.0.1:3003/api/v1/accounts/1/contacts/29/contact_inboxes' --header 'Content-Type: application / json' --header 'api_access_token: NQfhAnM58ThYybXV119BYrLP' --data-raw **'{"inbox_id": "2"}'**

Thank you all
